docs/cli-arguments.md

```diff
@@ -25,34 +25,34 @@ spec:
-      --port int                         Port to expose metrics on. (default 8080)
-      --resources string                 Comma-separated list of Resources to be enabled. Defaults to "certificatesigningrequests,configmaps,cronjobs,daemonsets,deployments,endpoints,horizontalpodautoscalers,ingresses,jobs,leases,limitranges,mutatingwebhookconfigurations,namespaces,networkpolicies,nodes,persistentvolumeclaims,persistentvolumes,poddisruptionbudgets,pods,replicasets,replicationcontrollers,resourcequotas,secrets,services,statefulsets,storageclasses,validatingwebhookconfigurations,volumeattachments"
-      --shard int32                      The instances shard nominal (zero indexed) within the total number of shards. (default 0)
-      --skip_headers                     If true, avoid header prefixes in the log messages
-      --skip_log_headers                 If true, avoid headers when opening log files
-      --stderrthreshold severity         logs at or above this threshold go to stderr (default 2)
-      --telemetry-host string            Host to expose kube-state-metrics self metrics on. (default "0.0.0.0")
-      --telemetry-port int               Port to expose kube-state-metrics self metrics on. (default 8081)
-      --total-shards int                 The total number of shards. Sharding is disabled when total shards is set to 1. (default 1)
-  -v, --v Level                          number for the log level verbosity
-      --version                          kube-state-metrics build version information
-      --vmodule moduleSpec               comma-separated list of pattern=N settings for file-filtered logging
+      --add_dir_header                    If true, adds the file directory to the header of the log messages
+      --alsologtostderr                   log to standard error as well as files
+      --apiserver string                  The URL of the apiserver to use as a master
+      --enable-gzip-encoding              Gzip responses when requested by clients via 'Accept-Encoding: gzip' header.
+  -h, --help                              Print Help text
+      --host string                       Host to expose metrics on. (default "0.0.0.0")
+      --kubeconfig string                 Absolute path to the kubeconfig file
+      --labels-metric-allow-list string   Allows to pass a list of additional Kubernetes label keys that will be used in the resource' labels metric. By default the metric contains only name and namespace labels. To include additional labels provide a list of resource names in their plural form and Kubernetes label keys you would like to allow for them (Example: '=namespaces=["k8s-label-1","k8s-label-n",...],pods=["app"],...)'
```

Comment:
Adjusted the help text, are we happy with flag name?
> --labels-metric-allow-list

My thinking for annotation it would be:
> --annotations-metric-label-allow-list

So might need a change here as well? 🤔  It gets confusing as its label metrics for label allow list. 🤯

Reply:
Flag name(s) sounds good to me.

I wonder if at some point, we want to consider reading arguments from a config file for all settings. I don't think we'll hit max cmdline input length, but let's assume someone is very verbose ;)
 It might also be easier to define it there in a structured way?
```
labels-metric-allow-list:
  - namespaces:
    - k8s-label1
    - k8s-label2
  - pods
    - app   
```

Reply:
Good point, I think if that happens we can add a flag called --labels-metric-allow-list-configfile that inputs a file, or one for both labels and annotations. Right now the parser makes certain assumptions (as it did before), I would say as we did not have a reading from a file until now I would not add it as part of this PR, but if there is something users face in the future, definitely!

Reply:
I agree, I was thinking  about a generic `--config` flag to do configuration for everything kube-state-metrics related.

Reply:
I see, okay if that is what you need, and you have use case, please feel free to open a feature request w and we can discuss there :)